src/focus-ring.js

```diff
@@ -74,11 +80,13 @@ document.addEventListener('DOMContentLoaded', function() {
    * opening a menu or dialog.
    */
   function onKeyDown() {
-    hadKeyboardEvent = true;
     // `activeElement` defaults to document.body if nothing focused,
     // so check the active element is actually focused.
-    if (matches(document.activeElement, ':focus'))
-      addFocusRingClass(document.activeElement);
+    var activeElement = document.activeElement;
+    if (activeElement.tagName == 'BODY')
+      return;
+
+    hadKeyboardEvent = true;
```

Comment:
Right: so at the point where this fires, `activeElement` is the element which is about to _lose_ focus, isn't it? So why _not_ set `hadKeyboardEvent` based on that?